AppendBlobReader/src/appendblobreader.json

```diff
@@ -132,63 +149,80 @@
+                            "apiVersion": "2022-04-01",
+                            "name": "[variables('consumer_roleGuid')]",
+                            "scope": "[variables('roleScope')]",
+                            "properties": {
+                                "roleDefinitionId": "[variables('roleDefinitionId')]",
+                                "principalId": "[reference(resourceId(variables('resourceGroupName'),'Microsoft.Web/sites', parameters('sites_blobreaderconsumer_name')), '2022-03-01', 'Full').Identity.principalId]"
+                            }
+                        },
+                        {
+                            "type": "Microsoft.Authorization/roleAssignments",
+                            "apiVersion": "2022-04-01",
+                            "name": "[variables('appendBlobProcessor_roleGuid')]",
+                            "scope": "[variables('roleScope')]",
                             "properties": {
                                 "roleDefinitionId": "[variables('roleDefinitionId')]",
-                                "principalId": "[reference(resourceId('Microsoft.Web/sites', parameters('sites_blobreaderconsumer_name')), '2022-09-01', 'Full').Identity.principalId]",
-                                "scope": "[variables('roleScope')]"
+                                "principalId": "[reference(resourceId(variables('resourceGroupName'),'Microsoft.Web/sites', parameters('sites_appendBlobProcessor_name')), '2022-03-01', 'Full').Identity.principalId]"
                             }
                         },
                         {
-                            "name": "[parameters('eventGridTopicName')]",
-                            "type": "Microsoft.EventGrid/topics",
-                            "location": "[parameters('location')]",
-                            "apiVersion": "2023-12-15-preview"
+                            "type": "Microsoft.EventGrid/systemTopics",
+                            "apiVersion": "2022-06-15",
+                            "name": "[parameters('eventGridSystemTopicName')]",
+                            "location": "[parameters('StorageAccountRegion')]",
+                            "properties": {
+                                "source": "[variables('roleScope')]",
+                                "topicType": "Microsoft.Storage.StorageAccounts"
+                            }
                         },
                         {
-                            "type": "Microsoft.Storage/storageAccounts/providers/eventSubscriptions",
-                            "name": "[concat(parameters('StorageAccountName'), '/Microsoft.EventGrid/', parameters('eventgridsubscription_name'))]",
-                            "location": "[parameters('location')]",
-                            "apiVersion": "2018-01-01",
+                            "type": "Microsoft.EventGrid/systemTopics/eventSubscriptions",
```

Comment:
why are we changing the eventsubscription definition ?

Reply:
As we updated the topic systemTopics
I updated eventsubscription definition as per systemTopics